### PR TITLE
Add Helper Functions

### PR DESCRIPTION
**Pull request by Sourav-Patteri**
Added a few helper functions, a function to handle form submission and a calculate quote function.

### CODE REVIEW

File: src/components/QuoteGenerator.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <form>
+  <form @submit.prevent="handleFormSubmit">
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:2]: Use single quotes wherever possible.

File: src/components/QuoteGenerator.vue [hunk 2]
```diff
@@ -153,7 +153,7 @@ import {
 const doors = ref<number>(1);
 const isPanelsOnlyOrder = ref<boolean>(false);
 const selectedDoorWidth = ref<string | null>(null);
-const selectedDoorHeight = ref<string | null>(null);
+const selectedDoorHeight = ref<string>("");
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:156]: Use single quotes.
Does removing `null` affect the select in any way?

File: src/components/QuoteGenerator.vue [hunk 3]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
...
+
+
+  // Call function to calculate quote
+  calculateQuote();
+
+};
+
+// Helper function to get the stamp pattern suffix
+function getStampPatternSuffix(stampPattern: string | null): string {
+  const suffixes: Record<string, string> = {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:213]: Could you define this as an Enum and move the same to `src/models/orders.ts`.
Or you could even extend the `KeyValuesSelects` interface to optionally hold this value as well.

File: src/components/QuoteGenerator.vue [hunk 4]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
...
+    SHXL: '4',
+    BCXL: '5',
+  };
+
+  return suffixes[stampPattern] || '';
+}
+
+// Helper function to get the panel color suffix
+function getPanelColorSuffix(panelColor: string | null): string {
+  const suffixes: Record<string, string> = {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:227]: Could you move this to `src/models/orders.ts`?

File: src/components/QuoteGenerator.vue [hunk 5]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
...
+  // Define the bulk panel prefix
+  const bulkPanelPrefix = bulkPanelsValue === true ? 'PN60' : 'PN65';
+
+  // Extract door height suffix
+  const doorHeightSuffix = doorHeightValue;
+
+  // Extract the first two characters of door width as door width suffix
+  const doorWidthSuffix = doorWidthValue.substring(0, 2);
+
+  const panelsTable: Record<string, Record<string, number>> = {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:259]: Could you move this to `src/models/orders.ts`, please?

File: src/store/store.ts
Comment (ksaswin) [inline src/store/store.ts:1]: The `store.ts` file can be removed if empty.

File: src/components/QuoteGenerator.vue [hunk 6]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
 function updateExtrasList(extra: string): void {
   selectedExtras.value.push(extra);
 }
+
+const handleFormSubmit = () => {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:199]: Could you declare the function in the following sytanx? (Just to keep it consistent in the project)
```ts
function handleFormSubmit(): void {
  // ...
}
```

File: src/components/QuoteGenerator.vue [hunk 7]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
...
+      partNumbers.push(partNumber);
+    }
+  });
+
+  console.log(`Your quote: Panel Part Numbers - ${partNumbers.join(', ')}.`)
+  return partNumbers.join(', ');
+
+};
+
+const calculateQuote = () => {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:307]: Could you declare the function in the following sytanx? (Just to keep it consistent in the project)
```ts
function calculateQuote(): void {
  // ...
}
```

File: src/components/QuoteGenerator.vue [hunk 8]
```diff
@@ -195,6 +195,119 @@ const displayDoorHeights = computed((): Array<string> => {
...
+
+const handleFormSubmit = () => {
+ 
+  console.log('Form submitted!');
+  console.log('Number of doors:', doors.value);
+  console.log('Is Panels Only Order:', isPanelsOnlyOrder.value);
+
+
+  // Call function to calculate quote
+  calculateQuote();
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:207]: If the handleFormSubmit does not have any other logic, `calculateQuote()` could be called directly from the form submit event

**Review by ksaswin — CHANGES_REQUESTED** [review]

**Comment by Sourav-Patteri** [comment]
Reviewed and resolved requested changes

**Pull request closed without merging**